### PR TITLE
Feature/rewrite slot picker

### PR DESCRIPTION
**Pull request by ptatarsk**
Done: 

- [x] Rewrite slot picker

- [x] Replace old picker on  rewrited

- [x] Change styles

### CODE REVIEW

**Comment by imbaniac** [comment]
Disable no-prop-types rule and add prop-types to all components
You can do it in a separate task

**Comment by ptatarsk** [comment]
> Disable no-prop-types rule and add prop-types to all components
> You can do it in a separate task

Done in Feature/add prop types pr. + add auto endOfLine rule.

**Review by imbaniac — CHANGES_REQUESTED** [review]

File: client/src/components/shared/SlotPicker/SlotPicker.js [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+  const handleMouseUpOrLeave = () => {
+    setPreSelectedItems([]);
+    setOptions({ ...options, isSelecting: false, isDeselect: false });
+    if (onSelectionEnd) {
+      onSelectionEnd(selectedItems);
+    }
+  };
+
+  const handleMouseDown = ({ target, clientX, clientY, target: { id } }) => {
+    setStartRectCoords({ ...startRectCoords, x: clientX, y: clientY });
```
[the remark this answers is not in the document]
Comment (ptatarsk) [reply client/src/components/shared/SlotPicker/SlotPicker.js]: It has more properties while ago. Fixed

File: client/src/components/shared/SlotPicker/SlotPicker.js [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+    if (target.classList.contains('selected')) {
+      setOptions({ ...options, isSelecting: true, isDeselect: true });
```
[the remark this answers is not in the document]
Comment (ptatarsk) [reply client/src/components/shared/SlotPicker/SlotPicker.js]: Fixed

**Pull request merged**